### PR TITLE
feat: limit functions exposed in indexing status API

### LINKED ISSUE #6: feat: Limit indexing status API
For `statusSever`,
1. extract graphql query from the received request
2. filter the query for supported root fields, 
  ```
  let supported_root_fields = [
  "indexingStatuses",
  "publicProofsOfIndexing",
  "entityChangesInBlock",
  "blockData",
  "cachedEthereumCalls",
  "subgraphFeatures",
  "apiVersions",
  ];
  ```
3. send the updated query request to `graph_node_status_endpoint`
4. return the response from graph node

### PR DESCRIPTION
**Pull request by hopeyen**
limit indexing status to serve supported root fields: `indexingStatuses, publicProofsOfIndexing, entityChangesInBlock, blockData, cachedEthereumCalls, subgraphFeatures, apiVersions`

So queries with root fields like `indexingStatusForCurrentVersion, indexingStatusForPendingVersion, indexingStatusesForSubgraphName, ...` should fail with bad request with a hint on the unsupported fields

Using `Regex` is brutal but get the job done while I didn't find similar functions in graphql crates like `async-graphql` and `graphql-parser` (also searched in `graphql-tools-rs` maintained by the guild.

Resolve #6

### CODE REVIEW

**Review by aasseman — DISMISSED** [review]
Simple and effective solution.

I think `graphql_parser` could have worked, but it would have been up to you to write a graph visitor to look for all the fields of the generated AST. More complex to write though, and probably worth it only if there's a need for a fancier GQL manipulations.

File: service/src/server/routes/status.rs [hunk 1]
```diff
@@ -18,13 +23,41 @@ pub async fn status_queries(
...
+        HashSet::from([
+            "indexingStatuses",
+            "publicProofsOfIndexing",
+            "entityChangesInBlock",
+            "blockData",
+            "cachedEthereumCalls",
+            "subgraphFeatures",
+            "apiVersions",
+        ])
+    }
```
[the remark this answers is not in the document]
Comment (hopeyen) [reply service/src/server/routes/status.rs]: updated to use `lazy_static` in [4392c10](https://github.com/graphops/indexer-service-rs/pull/55/commits/4392c1074210ecd4ddb343e562dffe08ad7bd54c) :)

**Comment by hopeyen** [comment]
@aasseman Thank you for the review! I've updated to use `lazy_static`. 


> I think graphql_parser could have worked, but it would have been up to you to write a graph visitor to look for all the fields of the generated AST

I agree, was well on my way in handling different GraphQL executions with `graphql_parser` and wanted to use it for a better error messages, but became complicated with the nested types and enums. perhaps we can submit a feature request instead of making it ourselves

**Review by aasseman — APPROVED** [review]
:shipit:

**Pull request merged**